.github/workflows/github-actions-on-delete-cleanup.yml

```diff
@@ -0,0 +1,30 @@
+name: Cleanup staging branch on delete
```

Comment:
This is not necessary since I have the github option to delete branches when they are merged set. Github automatically closes a PR and deletes its branch when merged. When the sync from public main to staging happens with a branches changes after a PR has been merged public, the branch already is automatically deleted. Same for -private. I hand tested this.

Reply:
This only works if the PR is merged. If it is abandoned or deleted for another reason it will be left on the staging repository.